aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -248,24 +234,18 @@ void update(const KeyType& key, T& results) {
   } else if (lru_cache_limit) {
     auto it = engine_cache.find(key);
     if (it == engine_cache.end()) {
-      auto engine_cache_order_size = engine_cache_order.size();
-      auto engine_cache_size = engine_cache.size();
-      TORCH_INTERNAL_ASSERT(engine_cache_order_size == engine_cache_size, "CUDNN V8 LRU Cache Corrupted (list vs. map size mismatch). Please report a bug to PyTorch.");
-      if ((long) engine_cache_order_size >= lru_cache_limit) {
-        // need to perform eviction
-        TORCH_INTERNAL_ASSERT(engine_cache.find(engine_cache_order.front()) != engine_cache.end(), "CUDNN V8 LRU Cache Corrupted (eviction key not in map). Please report a bug to PyTorch.");
-        engine_cache.erase(engine_cache_order.front());
-        engine_cache_order.pop_front();
+      if ((long) engine_cache.size() >= lru_cache_limit) {
+        auto erase_count = engine_cache.erase(engine_cache_order.back());
+        TORCH_INTERNAL_ASSERT(erase_count == 1, "CUDNN V8 LRU Cache Corrupted (eviction key not in map). Please report a bug to PyTorch.");
+        engine_cache_order.pop_back();
       }
+      engine_cache_order.emplace_front(key);
+      engine_cache.emplace(key, std::make_pair(results, engine_cache_order.begin()));
```

Comment:
Could we move the `engine_cache.emplace` below the `else` branch? Otherwise it seems that we would only update it in the `it == engine_cache.end()` case.

I'll write a summary comment in the main PR thread on this issue.